### ISSUE TITLE
want to build: spinner + 2 XY joysticks + buttons (driving controller)

### ISSUE THREAD

**Issue by CO-Windler** [CraigB-spinner/Arcade-Spinner#1]
I want to build a small driving/steering wheel controller compatible particularly with "Hard Drivin" and "Pole Position" variants on MAME, those need an endless 360° steering wheel (i.e. spinner) and 2 analogue "pedals" (brake and accelerator) plus a couple of digital buttons. Particularly "Hard Drivin" additionally needs an analogue XY joystick as a stick shifter, so I would need basically hardware that combines 2 PC joysticks and a trackball/ball mouse (containing 2 rotary encoders).

The device shall not become a fullsize simulator cockpit with steering wheel, but a compact controller similar like the Colecovision "Turbo" steering wheel/"Expansion Module 2". As a yogi I am a floorsitter living in a chairfree home, so it shall not need actual foot pedals (although I own plenty of volume pedals from my music keyboard collection); instead it will likely have pressure sensitive buttons or levers.

Possibly I will mod one of those baby steering wheel toys like this.

_Toy Steering Wheel Controller - Mario Kart: Double Dash!!_
https://www.youtube.com/watch?v=iNuI0311-qA

In my electronics parts box I also have a tube radio tuning knob assembly with bearing and big flywheel that could become a nice spinner.

(I websearched plenty of commercially available gamepads for for the electronic part but found nothing suitable. Modern PC mice and trackballs have optical sensors those can not easily mechanically separate X and Y motion into 2 independent controls.)

Can the code be used to handle a spinner+2 analogue joysticks+buttons combination with USB connector for PC?

**Comment by CraigB-spinner** [comment]
Yes, Pro Micro has ability for two spinners or one trackball, 10 digital buttons, 4 analogues (2 joysticks or 4 pots)
Sparkfun had an analogue joystick mouse project.
Warning don't overload the rotary encoder bearing with a heavy wheel or you will be buying twice. Think indirect drive through a gear, pulley, chain, or friction wheel. The steering wheel can be supported by its own bushing/bearings.
I used a 200mm (2.5" dia.) rubber/aluminum rotary encoder wheel.
